R/importFilterHypermutators.R

```diff
@@ -0,0 +1,101 @@
+# importFilterHypermutators.R
+
+#' Filters out hypermutators
+#'
+#' \code{importFilterHypermutators} identifies the number of SNV and CNA
+#' mutations for each sample and removes the samples from rMUT and rCNA
+#' datasets. Default threshold for removal is 400 mutations per sample.
+#'
+#' @param fNames A vector of local file names of rMUT and/or rCNA files
+#' @param dOut Directory to store output, defaults to getwd().
+#' @param xS Mutation threshold. 400 by default.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#'
+#' @examples
+#' \dontrun{
+#'     importFilterHypermutators(fNames, dOut, xS)
+#' }
+#'
+#' @export
+importFilterHypermutators <- function(fNames = c(),
+                                      dOut = getwd(),
+                                      xS = 400,
+                                      silent = FALSE,
+                                      writeLog = TRUE) {
+
+    ## VALIDATE PARAMS ########
+    # make sure that files actually exist
+    for (file in fNames) {
+        r <- .checkArgs(file, like = "FILE_E", checkSize = TRUE)
+        if (length(r) > 0) {
+            stop(sprintf("Local file name %s is invalid.", file))
+        }
+    }
+
+    # make sure that output directory is valid
+    r <- .checkArgs(fPath, like = "DIR", checkSize = TRUE)
+    if(length(r) > 0) {
+        stop(sprintf("Output directory %s is invalid.", dOut))
+    }
+
+    # make sure that xS is valid double
+    r <- .checkArgs(xS, like = "double", checkSize = TRUE)
+    if (length(r) > 0) {
+        stop(xS)
+    }
+
+
+    # set up hash
+    hashTable <- new.env(hash = TRUE)
+    totalRemovedSamples <- 0
+    totalSamples <- 0
+
+    ## COUNT MUTATIONS ########
+    # for each file in fNames,
+        # determine if it is rMUT or rCNA
+        # process rCNA files first
+            # if rCNA: read file, count # of gene-CNAs for each sample, store
```

Comment:
Rethinking this, the order in which they are processed doesn't really matter significantly. But it would be a good idea to store SNVs and CNAs separately in the hash, that way you can do a sanity check at the end whether the sample IDs actually overlapped. ... % of samples had bot, % of samples had only CNA, % of samples had only SNV, % of sampes had no change, % of samples exceeded threshold. We'll need to record the actual removed sample IDs. Logfile? Probably the best spot,  but then we should have a flag to suppress that, for cases where we anticipate large numbers of rejections.